sw/device/lib/dif/dif_spi_device.h

```diff
@@ -0,0 +1,243 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_SPI_DEVICE_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_SPI_DEVICE_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * Represents a signal edge type.
+ */
+typedef enum dif_spi_device_edge {
+  kDifSpiDeviceEdgePositive,
+  kDifSpiDeviceEdgeNegative,
+} dif_spi_device_edge_t;
+
+/**
+ * Represents a bit ordering.
+ */
+typedef enum dif_spi_device_bit_order {
+  kDifSpiDeviceBitOrderMsbToLsb,
+  kDifSpiDeviceBitOrderLsbToMsb,
+} dif_spi_device_bit_order_t;
```

Comment:
I know that the SPI tech spec uses this terminology, but aren't `MSBFirst` and `LSBFirst` used more commonly?

Reply:
I have no idea. Can someone comment on this?